### PR TITLE
refactor(ci): update go and add cve check

### PR DESCRIPTION
**Pull request by universal-itengineer**
## Description

Added CI weekly CVE check via trivy
Updated go version to `1.22.7` to prevent recent CVEs.
Remove unnecessary base go images
Updated base image alpine image to 3.17.10
Updated base image debian:bookworm-slim

## Why do we need it, and what problem does it solve?

Improved protection of images against various vulnerabilities
Improved ci
## What is the expected result?

All pods will be restarted. 
VMs will be migrated, if workload settings set to livemigration in other case you should manually restart them.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

### CODE REVIEW

File: .github/workflows/cve-weekly.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+      - name: Install Task
+        uses: arduino/setup-task@v2
+        with:
+          version: 3.37.2
+
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```
Comment (diafour) [inline .github/workflows/cve-weekly.yaml:58]: There is no `github.event.pull_request` for workflow_dispatch and schedule triggers.

File: .github/workflows/cve-weekly.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        run: |
+          echo $REPORT_FILE_NAME
+          REPORT_FILE_NAME=$REPORT_FILE_NAME task cve:scan
+
+      - name: Upload Trivy report as artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: trivy-weekly-${{ env.REPORT_FILE_NAME }}
+          path: ${{ env.REPORT_FILE_NAME }}
+          retention-days: 7
```
Comment (diafour) [inline .github/workflows/cve-weekly.yaml:72]:
```suggestion
          retention-days: 14
```
Make it at least 2 weeks, as we definitely want to access current and previous reports to compare.

File: Taskfile.yaml [hunk 3]
```diff
@@ -109,3 +110,16 @@ tasks:
...
+      - |
+        curl -LO https://github.com/aquasecurity/trivy/releases/download/v{{.TRIVY_VERSION}}/trivy_{{.TRIVY_VERSION}}_Linux-64bit.tar.gz
+        tar -zxvf trivy_{{.TRIVY_VERSION}}_Linux-64bit.tar.gz -C /opt/
+        mv /opt/trivy /usr/local/bin/trivy
+        chmod +x /usr/local/bin/trivy
+
+  cve:scan:
+    cmds:
+      - task cve:bin
+      - tools/cve/scan-main.sh {{.REPORT_FILE_NAME}} pr375
```
Comment (diafour) [inline Taskfile.yaml:125]: Should it be 'main'?  Don't forget to change pr375 before merging.

File: api/client/examples/list-resources/go.mod [hunk 4]
```diff
@@ -1,22 +1,25 @@
...
 require (
 	github.com/davecgh/go-spew v1.1.1 // indirect
 	github.com/emicklei/go-restful/v3 v3.11.0 // indirect
 	github.com/go-logr/logr v1.4.1 // indirect
 	github.com/go-openapi/jsonpointer v0.19.6 // indirect
 	github.com/go-openapi/jsonreference v0.20.2 // indirect
 	github.com/go-openapi/swag v0.22.3 // indirect
-	github.com/gogo/protobuf v1.3.2 // indirect
-	github.com/golang/protobuf v1.5.4 // indirect
 	github.com/google/gnostic-models v0.6.8 // indirect
```
Comment (diafour) [inline api/client/examples/list-resources/go.mod:23]: Why separate require? Is it after `go mod tidy` or you change it accidentally?

File: images/libguestfs/werf.inc.yaml [hunk 5]
```diff
@@ -26,9 +26,10 @@ shell:
     libguestfs==1.48.4-alt2:p10+307399.200.6.1 \
     guestfs-tools==1.48.2-alt1:p10+307399.300.6.1 \
     libvirt-daemon-driver-qemu==9.7.0-alt2.p10.2:p10+343223.200.3.1 \
-    qemu-kvm-core==8.2.4-alt0.p10.1:p10+350268.100.5.1 \
+    qemu-kvm-core==8.2.6-alt0.p10.1:p10+353259.100.4.1 \
     selinux-policy-alt==0.0.52-alt1:sisyphus+260794.100.1.1 \
-    libaltselinux==0.1.0-alt1
+    libaltselinux==0.1.0-alt1 \
+    python3-dev==3.9.18-alt1:p10+340781.400.5.1
```
Comment (diafour) [inline images/libguestfs/werf.inc.yaml:32]: Do we need python3-**dev**? I think only python3 should be updated.

File: .github/workflows/cve-weekly.yaml (hunk 1 above)
Comment (universal-itengineer) [reply .github/workflows/cve-weekly.yaml:58]: Just for the test

File: Taskfile.yaml (hunk 3 above)
Comment (universal-itengineer) [reply Taskfile.yaml:125]: Yes, i remember this)

File: api/client/examples/list-resources/go.mod (hunk 4 above)
Comment (universal-itengineer) [reply api/client/examples/list-resources/go.mod:23]: Is it after go mod tidy, I'll try move them back

File: Taskfile.yaml [hunk 6]
```diff
@@ -109,3 +110,16 @@ tasks:
...
+    cmds:
+      - |
+        curl -LO https://github.com/aquasecurity/trivy/releases/download/v{{.TRIVY_VERSION}}/trivy_{{.TRIVY_VERSION}}_Linux-64bit.tar.gz
+        tar -zxvf trivy_{{.TRIVY_VERSION}}_Linux-64bit.tar.gz -C /opt/
+        mv /opt/trivy /usr/local/bin/trivy
+        chmod +x /usr/local/bin/trivy
+
+  cve:scan:
+    cmds:
+      - task cve:bin
```
Comment (diafour) [inline Taskfile.yaml:124]:
```suggestion
      - task: cve:bin
```

Comment (universal-itengineer) [reply Taskfile.yaml]: Fixed

File: tools/cve/scan-main.sh [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+    echo "⭐ ==============================================================================================================="
+    echo "name: ${name}"
+    echo "image: ${image}"
+    echo "=================================================================================================================="
+
+    trivy image ${image} -f table
+
+  done <<< $(echo ${images} | jq -c .)
+} > "${REPORT_FILE_NAME}"
```
Comment (diafour) [inline tools/cve/scan-main.sh:41]: Optimize jq calls:

```suggestion
# Prepare images digests in form of "image_name image_sha256_digest".
images_digests=$(crane export dev-registry.deckhouse.io/sys/deckhouse-oss/modules/virtualization:${module_tag}  - | tar -Oxf - images_digests.json | jq -r 'to_entries[] | .name + " " + .value')

{
  while read name digest; do
    image="dev-registry.deckhouse.io/sys/deckhouse-oss/modules/virtualization@${digest}"

    if [[ ${name} =~ Builder|Artifact ]]; then
      continue
    fi

    echo "⭐ ==============================================================================================================="
    echo "name: ${name}"
    echo "image: ${image}"
    echo "=================================================================================================================="

    trivy image ${image} -f table

  done <<< "${images_digests}"
} > "${REPORT_FILE_NAME}"
```

File: images/virt-launcher/werf.inc.yaml [hunk 8]
```diff
@@ -296,7 +304,7 @@ shell:
     xorriso==1.5.6-alt1:p11+348905.100.1.3 \
     dosfstools==4.2-alt3:sisyphus+344264.100.1.1 \
     mtools==1:4.0.43-alt1:sisyphus+317197.100.1.1 \
-    qemu-img==8.2.4-alt1:p11+350544.100.2.1 \
+    qemu-img==9.0.2-alt2:p11+353258.200.3.2 \
```
Comment (diafour) [inline images/virt-launcher/werf.inc.yaml:307]:
```suggestion
    qemu-img==8.2.4-alt1:p11+350544.100.2.1 \
```

No need to rebuild this image, let's revert qemu-img versin to use cached image.

**Review by diafour — APPROVED** [review]

**Pull request merged**